src/platform/packages/shared/react/kibana_context/render/render_provider.tsx

```diff
@@ -19,8 +19,9 @@ import { KibanaErrorBoundary, KibanaErrorBoundaryProvider } from '@kbn/shared-ux
 export type KibanaRenderContextProviderProps = Omit<KibanaRootContextProviderProps, 'globalStyles'>;
 
 /**
- * The `KibanaRenderContextProvider` provides the necessary context for an out-of-current
- * React render, such as using `ReactDOM.render()`.
+ * The `KibanaRenderContextProvider` provides the necessary context for an out-of-current React render, such as using `ReactDOM.render()`.
+ *
+ * @internal Use RenderContextService.addContext from the CoreStart contract instead of consuming this directly.
```

Comment:
nit:

```suggestion
 * @deprecated Use RenderContextService.addContext from the CoreStart contract instead of consuming this directly.
```

But `internal` works too, it's just that most IDEs should strikethrough the text if we use `@deprecated`

Reply:
I think we will keep asking teams to use this to wrap around components in test files, which could be the best thing to keep doing in test files. That is happening already in a few places. So I'll leave this as-is. :)